core/components/molecules/keyValuePair/KeyElement.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import { IconProps } from '@/index.type';
+import { Icon, Text } from '@/index';
+import classNames from 'classnames';
+import { BaseProps, extractBaseProps } from '@/utils/types';
+
+type IconPosition = 'left' | 'right';
+
+export interface KeyElementProps extends BaseProps {
+  /**
+   * React Element to be added inside `KeyElement`
+   */
+  children?: React.ReactNode;
+  /**
+   * Specify label to be displayed in `KeyElement`
+   */
+  label?: React.ReactText;
+  /**
+   * Specify Icon name to displayed in `KeyElement`
+   */
+  icon?: string;
+  /**
+   * List of icon options
+   *
+   * <pre className="DocPage-codeBlock">
+   *  IconProps: {
+   *    name?: string;
+   *    size: number;
+   *    type?: 'rounded' | 'outlined';
+   *    appearance?: IconAppearance;
+   *    onClick?: (e: React.MouseEvent<HTMLElement>) => void;
+   *    onKeyDown?: (e: React.KeyboardEvent<HTMLElement>) => void;
+   *    children?: React.ReactNode;
+   *    tabIndex?: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>['tabIndex'];
+   *  }
+   * </pre>
+   */
+  iconOptions?: IconProps;
+  /**
+   * Align icon left or right
+   */
+  iconAlign?: IconPosition;
+}
+
+export const KeyElement = (props: KeyElementProps) => {
+  const { children, icon, iconOptions, iconAlign, label, className } = props;
+
+  const baseProps = extractBaseProps(props);
+
+  if (children) {
+    return (
+      <dt data-test="DesignSystem-KeyValuePair-KeyElement" {...baseProps}>
+        {children}
+      </dt>
+    );
+  }
```

Comment:
Move this block to end of variable definitions, keep variable definitions on top.